internal/fwserver/attr_value.go

```diff
@@ -35,11 +36,15 @@ func coerceMapValue(schemaPath path.Path, value attr.Value) (types.Map, diag.Dia
 	return m, nil
 }
 
-func coerceObjectValue(schemaPath path.Path, value attr.Value) (types.Object, diag.Diagnostics) {
-	object, ok := value.(types.Object)
+// coerceObjectValue no longer coerces value to types.Object, but performs a type assertion to verify that value fills
+// attr.ValueWithAttrs interface.
+func coerceObjectValue(schemaPath path.Path, value attr.Value) (attr.Value, diag.Diagnostics) {
```

Comment:
Function signature could be altered to return `attr.ValueWithAttrs` so that type assertions are not required elsewhere.

Function name should also change to something like `coerceValueWithAttrs`.